src/main/java/eod/card/collection/Deck.java

```diff
@@ -43,6 +44,24 @@ public Deck(ArrayList<Card> cards, boolean shuffle) {
 
     }
 
+    public Card[] draw(Class<? extends Card> cardType, int count) {
+        List<Card> canDraw = cards.stream().filter(card -> card.cardTypeEquals(cardType)).collect(Collectors.toList());
+        int n = Math.min(canDraw.size(), count);
+
+        Card[] drew = new Card[n];
+
+        for(int i = 0;i < n;i++) {
+            drew[i] = canDraw.remove(0);
+            cards.remove(drew[i]);
```

Comment:
Not sure if it is the same reference, tests needed

Reply:
solved